source/logger.c

```diff
@@ -131,7 +134,15 @@ static void s_ring_buffer_mem_release(struct aws_allocator *allocator, void *ptr
         .len = 0,
     };
     struct aws_ring_buffer *buffer = allocator->impl;
-    aws_ring_buffer_release(buffer, &buf);
+
+    /* see if the memory comes from the ring buffer */
+    if (aws_ring_buffer_buf_belongs_to_pool(buffer, &buf)) {
+        aws_ring_buffer_release(buffer, &buf);
+        return;
```

Comment:
random style thing, instead of returning here, just put the `aws_meme_release` call in an `else` block.